@xen-orchestra/web/src/components/hosts/HostsTable.vue

```diff
@@ -0,0 +1,231 @@
+                </UiTagsList>
+              </template>
+            </span>
+          </td>
+        </tr>
+      </template>
+    </VtsDataTable>
+    <VtsStateHero v-if="searchQuery && filteredHosts.length === 0" format="table" type="no-result" size="medium">
+      {{ t('no-result') }}
+    </VtsStateHero>
+    <div class="table-actions">
+      <UiTopBottomTable :selected-items="0" :total-items="0">
+        <UiTablePagination v-bind="paginationBindings" />
+      </UiTopBottomTable>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useXoPifCollection } from '@/remote-resources/use-xo-pif-collection'
+import type { XoHost } from '@/types/xo/host.type'
+import type { IconName } from '@core/icons'
+import VtsDataTable from '@core/components/data-table/VtsDataTable.vue'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import VtsObjectIcon from '@core/components/object-icon/VtsObjectIcon.vue'
+import VtsStateHero from '@core/components/state-hero/VtsStateHero.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiObjectLink from '@core/components/ui/object-link/UiObjectLink.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTablePagination from '@core/components/ui/table-pagination/UiTablePagination.vue'
+import UiTag from '@core/components/ui/tag/UiTag.vue'
+import UiTagsList from '@core/components/ui/tag/UiTagsList.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import { usePagination } from '@core/composables/pagination.composable'
+import { useRouteQuery } from '@core/composables/route-query.composable'
+import useMultiSelect from '@core/composables/table/multi-select.composable'
+import { useTable } from '@core/composables/table.composable'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import { HOST_POWER_STATE } from '@vates/types'
+import { noop } from '@vueuse/shared'
+import { computed, ref } from 'vue'
+import { useI18n } from 'vue-i18n'
+
+const { hosts } = defineProps<{
+  hosts: XoHost[]
+  hostReady: boolean
+  hasError: boolean
+}>()
+
+const hostsId = computed(() => hosts.map(host => host.id))
+
+const { t } = useI18n()
+const selectedHostId = useRouteQuery('id')
+const { pifsByHost, arePifsReady } = useXoPifCollection()
+const { selected, areAllSelected } = useMultiSelect(hostsId)
+
+const toggleSelect = () => {
+  selected.value = selected.value.length === 0 ? hostsId.value : []
+}
+
+const searchQuery = ref('')
+
+const filteredHosts = computed(() => {
+  const searchTerm = searchQuery.value.trim().toLocaleLowerCase()
+
+  if (!searchTerm) {
+    return hosts
+  }
+
+  return hosts.filter(host => Object.values(host).some(value => String(value).toLocaleLowerCase().includes(searchTerm)))
+})
+
+const { visibleColumns, rows } = useTable('hosts', filteredHosts, {
```

Comment:
```suggestion
})

const getIpAddresses = (host: XoHost) => {
  const pifs = pifsByHost.value.get(host.id)
  if (!pifs) {
    return []
  }

  return pifs.reduce<string[]>((acc, pif) => {
    if (pif.ip) {
      acc.push(pif.ip)
    }

    if (pif.ipv6) {
      acc.push(...pif.ipv6.filter(ip => ip))
    }

    return acc
  }, [])
}

const { visibleColumns, rows } = useTable('hosts', filteredHosts, {
```

we can do something like that ( I think it can be simplify and you should extact it in a composable too)

And we need to display all IPs in the table (V4 and V6) -> Yes , isn't the case in the figma but Clemence needs to change it like the vms tab